packages/ai-ide/src/browser/ai-configuration/mcp-configuration-widget.tsx

```diff
@@ -0,0 +1,328 @@
+// *****************************************************************************
+// Copyright (C) 2024 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { ReactWidget } from '@theia/core/lib/browser';
+import { inject, injectable, postConstruct } from '@theia/core/shared/inversify';
+import * as React from '@theia/core/shared/react';
+import { HoverService } from '@theia/core/lib/browser/hover-service';
+import { MCPFrontendNotificationService, MCPFrontendService, MCPServerDescription, MCPServerStatus } from '@theia/ai-mcp/lib/common/mcp-server-manager';
+import { MessageService } from '@theia/core';
+
+@injectable()
+export class AIMCPConfigurationWidget extends ReactWidget {
+
+    static readonly ID = 'ai-mcp-configuration-container-widget';
+    static readonly LABEL = 'MCP Servers';
+
+    servers: MCPServerDescription[] = [];
+    expandedTools: Record<string, boolean> = {};
+
+    @inject(MCPFrontendService)
+    protected readonly mcpFrontendService: MCPFrontendService;
+
+    @inject(MCPFrontendNotificationService)
+    protected readonly mcpFrontendNotificationService: MCPFrontendNotificationService;
+
+    @inject(HoverService)
+    protected readonly hoverService: HoverService;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    @postConstruct()
+    protected init(): void {
+        this.id = AIMCPConfigurationWidget.ID;
+        this.title.label = AIMCPConfigurationWidget.LABEL;
+        this.title.closable = false;
+        this.toDispose.push(this.mcpFrontendNotificationService.onDidUpdateMCPServers(async () => {
+            this.loadServers();
+        }));
+        this.loadServers();
+    }
+
+    protected async loadServers(): Promise<void> {
+        const serverNames = await this.mcpFrontendService.getServerNames();
+        const promises = serverNames.map(name => this.mcpFrontendService.getServerDescription(name));
```

Comment:
promises is very generic, maybe use something like: descriptionPromises or just merge this and the next line?